rasa/shared/utils/io.py

```diff
@@ -310,8 +310,8 @@ def env_var_constructor(loader, node):
         """Process environment variables found in the YAML."""
         value = loader.construct_scalar(node)
         expanded_vars = os.path.expandvars(value)
-        if "$" in expanded_vars:
-            not_expanded = [w for w in expanded_vars.split() if "$" in w]
+        not_expanded = [w for w in expanded_vars.split() if w.startswith("$") and w in value]
```

Comment:
Should we also check for `%` which is used for env variables on Windows?

Reply:
We don't support it — for example, the regex (`env_var_pattern`) looks only for variables that start with `$`.
Currently all the texts that contain `%` will be left untouched and I think it's the correct behaviour.